### PR TITLE
[Vectorized][Bug] fix get wrong result when select random column && fix get wrong has_null_tag

### LINKED ISSUE #7727: [Vectorized][Bug] fix get wrong result when select random column && fix get wrong `has_null_tag`
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

vectorized

### What's Wrong?

fix get wrong result when select random column

### What You Expected?

fix

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

Issue Number: close #7727

## Problem Summary:

Describe the overview of changes.

## Checklist(Requried)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by spaces-X** [comment]
the same issue occurred to me

LGTM

File: be/src/vec/olap/block_reader.cpp [hunk 1]
```diff
@@ -84,7 +84,8 @@ void BlockReader::_init_agg_state() {
 
     auto& tablet_schema = tablet()->tablet_schema();
     for (auto idx : _agg_columns_idx) {
-        FieldAggregationMethod agg_method = tablet_schema.column(idx).aggregation();
+        FieldAggregationMethod agg_method =
```
Comment (zuochunwei) [inline be/src/vec/olap/block_reader.cpp:87]: use hash_map is not a good idea

File: be/src/vec/olap/block_reader.cpp [hunk 2]
```diff
@@ -84,7 +84,8 @@ void BlockReader::_init_agg_state() {
...
+                tablet_schema.column(_real_columns_idx[idx]).aggregation();
```
Comment (zuochunwei) [inline be/src/vec/olap/block_reader.cpp:88]: _real_columns_idx[idx] 
can be replaced by 
read_params.origin_return_columns->at(_return_columns_loc[idx])

File: be/src/vec/olap/block_reader.cpp [hunk 3]
```diff
@@ -130,6 +131,7 @@ OLAPStatus BlockReader::init(const ReaderParams& read_params) {
                     _normal_columns_idx.emplace_back(j);
                 } else {
                     _agg_columns_idx.emplace_back(j);
+                    _real_columns_idx[j] = cid;
```
Comment (spaces-X) [inline be/src/vec/olap/block_reader.cpp:134]: I think it is better to named `_real_agg_columns_idx`  or `_origin_agg_columns_idx`

File: be/src/vec/olap/block_reader.cpp (hunk 2 above)
Comment (BiteTheDDDDt) [reply be/src/vec/olap/block_reader.cpp]: Okay,I modified this.

Comment (spaces-X) [reply be/src/vec/olap/block_reader.cpp]: `read_params` has been gone after init().

I think it is better to named `_real_agg_columns_idx`  or `_origin_agg_columns_idx`

**Comment by zuochunwei** [comment]
LGTM

**Pull request merged**